quixstreams/state/windows.py

```diff
@@ -0,0 +1,355 @@
+        self._dataframe = dataframe
+        self._name = name
+
+    def _get_name(self, func_name: str) -> str:
+        return self._name or f"tumbling_window_{self._duration}_{func_name}"
+
+    def sum(self) -> "TumblingWindow":
+        """
+        Configure the tumbling window to aggregate data by summing up values within each window period.
+
+        :return: TumblingWindow instance configured to perform sum aggregation.
+        """
+        name = self._get_name(func_name="sum")
+
+        def func(start, end, timestamp, value: Any, state: WindowedTransactionState):
+            current_value = state.get_window(start=start, end=end) or 0
+            updated_value = current_value + value
+
+            state.update_window(start, end, timestamp=timestamp, value=updated_value)
+            return updated_value
+
+        return TumblingWindow(
+            duration=self._duration,
+            grace=self._grace,
+            name=name,
+            func=func,
+            dataframe=self._dataframe,
+        )
+
+    def count(self) -> "TumblingWindow":
+        """
+        Configure the tumbling window to aggregate data by counting the number of records within each window period.
+
+        :return: TumblingWindow instance configured to perform record count.
+        """
+        name = self._get_name(func_name="count")
+
+        def func(start, end, timestamp, _: Any, state: WindowedTransactionState):
+            current_value = state.get_window(start=start, end=end) or 0
+            updated_value = current_value + 1
+
+            state.update_window(start, end, timestamp=timestamp, value=updated_value)
+            return updated_value
+
+        return TumblingWindow(
+            duration=self._duration,
+            grace=self._grace,
+            name=name,
+            func=func,
+            dataframe=self._dataframe,
+        )
+
+    def mean(self) -> "TumblingWindow":
+        """
+        Configure the tumbling window to aggregate data by calculating the mean of the values within each window period.
+
+        :return: TumblingWindow instance configured to calculate the mean of the data values.
+        """
+        name = self._get_name(func_name="mean")
+
+        def func(start, end, timestamp, value: Any, state: WindowedTransactionState):
+            current_window_value = state.get_window(start=start, end=end) or (0.0, 0)
+
+            agg = current_window_value[0] + value
+            count = current_window_value[1] + 1
+
+            updated_window_value = (agg, count)
+
+            state.update_window(
+                start, end, timestamp=timestamp, value=updated_window_value
+            )
+            return agg / count
+
+        return TumblingWindow(
+            duration=self._duration,
+            grace=self._grace,
+            name=name,
+            func=func,
+            dataframe=self._dataframe,
+        )
+
+    def reduce(self, reduce_func: Callable[[Any, Any], Any]) -> "TumblingWindow":
```

Comment:
For reduce function I think we need to have a dedicated `default` or `initializer` param similar to https://docs.python.org/3/library/functools.html#functools.reduce

Using the first window item as initializer is not exactly correct, because the expected return type of the window can be different from the incoming value type.
E.g. your incoming values are numbers, but you need to produce a dictionary with "min", "max", "sum" keys. If you simply return a number, it will not work